internal/transport/transport_test.go

```diff
@@ -2656,10 +2656,33 @@ func TestConnectionError_Unwrap(t *testing.T) {
 	}
 }
 
-// Test that in the event of a graceful client transport shutdown, i.e.,
+// TestClientSendsAGoAwayFrame verifies that in the event of a graceful client transport shutdown, i.e.,
 // clientTransport.Close(), client sends a goaway to the server with the correct
 // error code and debug data.
 func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
+	errorCh := createClientServerLoggingGoAway(t)
+	err := <-errorCh
+	if err != nil {
+		t.Errorf("Error receiving the GOAWAY frame: %v", err)
```

Comment:
if error happens here its not clear whether it happened at client or server. It would be best if you keep the client and server startup code separate or just have the same code duplicated in the test